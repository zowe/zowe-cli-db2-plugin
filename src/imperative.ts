/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { IImperativeConfig } from "@brightside/imperative";
import { Constants } from "./";

const config: IImperativeConfig = {
    name: Constants.PROD_NAME,
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    rootCommandDescription: Constants.DESCRIPTION,
    productDisplayName: Constants.DISPLAY_NAME,
    pluginHealthCheck: "./lib/HealthCheck.handler",
    profiles: [
        {
            type: "db2",
            schema: {
                type: "object",
                title: "Db2 Profile",
                description: "A profile for interaction with Db2 for the z/OS region",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: {
                            name: "host",
                            aliases: ["H"],
                            description: "The Db2 server host name",
                            type: "string",
                        },
                    },
                    port: {
                        type: "number",
                        optionDefinition: {
                            name: "port",
                            aliases: ["P"],
                            description: "The Db2 server port number",
                            type: "number",
                        },
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "user",
                            aliases: ["u"],
                            description: "The Db2 user ID (may be the same as the TSO login)",
                            type: "string",
                        },
                    },
                    password: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "password",
                            aliases: ["pass", "pw"],
                            description: "The Db2 password (may be the same as the TSO password)",
                            type: "string",
                        },
                    },
                    database: {
                        type: "string",
                        optionDefinition: {
                            name: "database",
                            aliases: ["d"],
                            description: "The name of the database",
                            type: "string",
                        },
                    },
                    sslFile: {
                        type: "string",
                        optionDefinition: {
                            name: "ssl-file",
                            aliases: ["s"],
                            description: "Path to an SSL Certificate file",
                            type: "string",
                        },
                    },
                },
            },
        }
    ],
};

export = config;
