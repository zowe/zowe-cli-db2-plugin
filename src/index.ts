/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

export * from "./Constants";
export * from "./api/ConnectionString";
export * from "./api/doc/IDB2Column";
export * from "./api/doc/IDB2Parameter";
export * from "./api/doc/IDB2Response";
export * from "./rest/session/doc/IDB2Session";
export * from "./cli/DB2Session";
export * from "./cli/DB2BaseHandler";
export * from "./rest/session/Session";
export * from "./api/DB2Constants";
export * from "./api/SessionValidator";
export * from "./api/ExecuteSQL";
export * from "./api/ExportTable";
export * from "./api/ExportTableSQL";
export * from "./api/CallSP";
