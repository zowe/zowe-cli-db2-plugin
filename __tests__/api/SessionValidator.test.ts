/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { SessionValidator, IDB2Session } from "../../src";
import * as C from "../Db2TestConstants";

const SESSION_GOOD: IDB2Session = {
    database: C.DATABASE_NAME,
    hostname: C.HOST_NAME,
    port: C.PORT,
    user: C.USER_NAME,
    password: C.PASSWORD,
};

const SESSION_NO_HOST: IDB2Session = {
    database: C.DATABASE_NAME,
    hostname: undefined,
    port: C.PORT,
    user: C.USER_NAME,
    password: C.PASSWORD,
};

const SESSION_NO_PORT: IDB2Session = {
    database: C.DATABASE_NAME,
    hostname: C.HOST_NAME,
    port: null,
    user: C.USER_NAME,
    password: C.PASSWORD,
};

const SESSION_NO_DB: IDB2Session = {
    database: "",
    hostname: C.HOST_NAME,
    port: C.PORT,
    user: C.USER_NAME,
    password: C.PASSWORD,
};

const SESSION_NO_USER: IDB2Session = {
    database: C.DATABASE_NAME,
    hostname: C.HOST_NAME,
    port: C.PORT,
    user: undefined,
    password: C.PASSWORD,
};

const SESSION_NO_PASS: IDB2Session = {
    database: C.DATABASE_NAME,
    hostname: C.HOST_NAME,
    port: C.PORT,
    user: C.USER_NAME,
    password: undefined,
};

describe("SessionValidator", () => {
    describe("validate", () => {
        it("should accept a valid DB2 session object", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_GOOD);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeUndefined();
        });

        it("should fail validating a null DB2 session object", () => {
            let error;
            try {
                SessionValidator.validate(null);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating an empty DB2 session object", () => {
            let error;
            const empty: any = {};
            try {
                SessionValidator.validate(empty);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating a DB2 session object with empty 'hostname' field", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_NO_HOST);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating a DB2 session object with empty 'port' field", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_NO_PORT);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating a DB2 session object with empty 'database' field", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_NO_DB);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating a DB2 session object with empty 'username' field", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_NO_USER);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });

        it("should fail validating a DB2 session object with empty 'password' field", () => {
            let error;
            try {
                SessionValidator.validate(SESSION_NO_PASS);
            }
            catch (e) {
                error = e;
            }
            expect(error).toBeDefined();
            expect(error.toString()).toMatchSnapshot();
        });
    });
});
