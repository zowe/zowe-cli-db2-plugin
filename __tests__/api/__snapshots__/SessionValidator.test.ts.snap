// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionValidator validate should fail validating a DB2 session object with empty 'database' field 1`] = `"Error: Expect Error: Required parameter 'database' must not be blank"`;

exports[`SessionValidator validate should fail validating a DB2 session object with empty 'hostname' field 1`] = `"Error: Expect Error: Imperative API Error - No DB2 server hostname was supplied."`;

exports[`SessionValidator validate should fail validating a DB2 session object with empty 'password' field 1`] = `"Error: Expect Error: Imperative API Error - No DB2 user password was supplied."`;

exports[`SessionValidator validate should fail validating a DB2 session object with empty 'port' field 1`] = `"Error: Expect Error: Imperative API Error - No DB2 server port number was supplied."`;

exports[`SessionValidator validate should fail validating a DB2 session object with empty 'username' field 1`] = `"Error: Expect Error: Imperative API Error - No DB2 User ID was supplied."`;

exports[`SessionValidator validate should fail validating a null DB2 session object 1`] = `"Error: Expect Error: Imperative API Error - No DB2 parameters were supplied."`;

exports[`SessionValidator validate should fail validating an empty DB2 session object 1`] = `"Error: Expect Error: Imperative API Error - No DB2 server hostname was supplied."`;
